src/main/kotlin/it/pagopa/checkout/authservice/mdcutilities/MDCFilter.kt

```diff
@@ -21,8 +21,8 @@ class MDCFilter : WebFilter {
         val path = request.uri.path
 
         // extract header values
-        val rptId =
-            headers.getFirst(HEADER_RPT_ID) ?: RequestTracingUtils.TracingEntry.RPT_ID.defaultValue
+        var rptId = headers.getFirst(HEADER_RPT_IDS)
+        if (rptId.isNullOrEmpty()) rptId = RequestTracingUtils.TracingEntry.RPT_ID.defaultValue
```

Comment:
```suggestion
        val rptIds = headers.getFirst(HEADER_RPT_IDS).let {
            if (!it.isNullOrEmpty()) {
                it
            } else {
                RequestTracingUtils.TracingEntry.RPT_ID.defaultValue
            }
        }
```

🚲  less compact but produce a constant val instead of using a variable value that need to be smart cast later

Reply:
Or we can just go back to the previous implementation? 
(val rptId = headers.getFirst(HEADER_RPT_ID) ?: RequestTracingUtils.TracingEntry.RPT_ID.defaultValue)